agent/api/containerstatus.go

```diff
@@ -22,6 +22,13 @@ const (
 	ContainerCreated
 	// ContainerRunning represents a container that has started
 	ContainerRunning
+	// ContainerResourcesProvisioned represents a container that has completed provisioning all of its
+	// resources. Non-internal containers (containers present in the task definition) transition to
```

Comment:
This notion of a state after running is confusing and not intuitive to me. Potentially re-working the states to `Created -> Started -> Running` where the second transition is a no-op for non-internal containers could help clear some of that confusion up.